### ISSUE TITLE
xSQLServerAlwaysOnAvailabilityGroupReplica: Hash table returned from get-method does not match schema

### ISSUE THREAD

**Issue by johlju** [PowerShell/xSQLServer#477]
**Details of the scenario you try and problem that is occurring:**
The hash table that is returned from the Get-method does not contain the following properties 

PrimaryReplicaSQLServer
PrimaryReplicaSQLInstanceName
EndpointHostName

Regarding `EndpointHostName`, the hash table return the read only property `SQLServerNetName`, are that actually suppose to be set to `EndpointHostName` instead?

The hash table do contain this property which is not part of the schema. These should be added as read only properties.

EndpointUrl
EndpointPort

**The DSC configuration that is using the resource (as detailed as possible):**
n/a

**Version of the Operating System, SQL Server and PowerShell the DSC Target Node is running:**
n/a

**Version of the DSC module you're using, or 'dev' if you're using current dev branch:**
Dev

**Comment by randomnote1** [comment]
@johlju, I'm going to take this and work on it.

**Comment by randomnote1** [comment]
Thanks for detailing the issue @johlju. I'm going to try and document why some of the supplied parameters are not included in the hashtable that is returned from Get-TargetResource.

- The parameters **PrimaryReplicaSQLServer** and **PrimaryReplicaSQLInstanceName** are not important in determining the current state of the Availability Replica. These are only required when creating or updating the replica.
- **EndpointHostName** is not included in the hash table because the **EndpointURL** is dynamically created using the supplied **EndpointHostName**.
- **EndpointHostName** is only set to the **SQLServerNetName** if **EndpointHostName** is not supplied.

**Comment by johlju** [comment]
@randomnote1 depending on issue https://github.com/PowerShell/DscResources/issues/318 we should either add these properties to the hash table or leave it as is (except for the missing read properties that should be added).

Setting this as in progress for the time being.

**Comment by randomnote1** [comment]
I'm ready to submit the PR for this. Just waiting on the decision!

**Comment by johlju** [comment]
Please submit the PR. It's an approvement. I meant that we can start to enforce the rule once the issue in DscResources are "approved"/correct. :)

**Comment by randomnote1** [comment]
Gotcha. PR submitted.
